File: gdscript/src/main/java/gdscript/psi/GdConstDeclSt.java
package gdscript.psi;

import com.intellij.psi.PsiElement;
import gdscript.psi.types.GdDocumented;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

public interface GdConstDeclSt extends GdStmt, GdDocumented {

  @Nullable
  GdAssignTyped getAssignTyped();

  @Nullable
  GdEndStmt getEndStmt();

  @Nullable
  GdExpr getExpr();

  @Nullable
  GdTyped getTyped();

  @Nullable
  GdVarNmi getVarNmi();

  @NotNull
  String getName();

  @NotNull
  String getReturnType();

  @Nullable
  PsiElement getReturnExpr();

}

File: gdscript/src/main/java/gdscript/psi/GdConstDeclSt.kt
package gdscript.psi

import com.intellij.psi.PsiElement
import gdscript.psi.types.GdDocumented

interface GdConstDeclSt : GdStmt, GdDocumented {
    val assignTyped: GdAssignTyped?

    val endStmt: GdEndStmt?

    val expr: GdExpr?

    val typed: GdTyped?

    val varNmi: GdVarNmi?

    fun getName(): String

    val returnType: String

    val returnExpr: PsiElement?
}
